python/replicate/pl_callback.py

```diff
@@ -0,0 +1,49 @@
+from copy import deepcopy
+from typing import Optional, Dict
+
+import replicate
+from pytorch_lightning.callbacks.base import Callback
+
+
+class ReplicateCallback(Callback):
+    def __init__(
+        self,
+        params: Dict,
+        monitor: str,
+        period: Optional[int] = 1,
+        save_weights_only: Optional[bool] = False,
+        mode: Optional[str] = 'min'
+    ):
+        super().__init__()
+        self.params = params
+        self.monitor = monitor
+        self.period = period
+        self.save_weights_only = save_weights_only
+        self.mode = 'minimize' if mode == 'min' else 'maximize'
+
+    def on_pretrain_routine_start(self, trainer, pl_module):
+        self.experiment = replicate.init(path='.', params=self.params)
+
+    def on_validation_end(self, trainer, pl_module):
```

Comment:
I'm not a PL user myself so this is maybe a stupid question, but does PL always do validation? Or are there uses of PL where the Replicate callback wouldn't actually call `self._save_model`?

Reply:
@andreasjansson There are definitely cases where PL doesn't do validation. e.g. you just write train code, not validation code.

For the second question, I am not sure.

Reply:
p.s. you should consider trying PL, it's a great framework.

Reply:
@andreasjansson Yes. I think I should add checks for that. I need to investigate PL's ModelCheckpoint in more scenarios. I am not personally a PL user sadly, I am just familiar with the repo from contributing. 
@turian if you have any suggestions on what else I should check for please let me know.